src/EditorFeatures/CSharpTest/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzerTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UsePatternCombinators;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UsePatternCombinators
+{
+    public class CSharpUsePatternCombinatorsDiagnosticAnalyzerTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        private static readonly ParseOptions CSharp9 = TestOptions.RegularPreview;
+
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new CSharpUsePatternCombinatorsDiagnosticAnalyzer(), new CSharpUsePatternCombinatorsCodeFixProvider());
+
+        private Task TestMissingInCSharp9Async(string initialMarkup)
+            => TestMissingAsync(initialMarkup, new TestParameters(CSharp9));
+
+        private Task TestInCSharp9Async(string initialMarkup, string expectedMarkup)
+            => TestInRegularAndScriptAsync(initialMarkup, expectedMarkup, parseOptions: CSharp9);
+
+        private static readonly string s_initialMarkup = @"
+using System;
+using System.Collections.Generic;
+class C
+{
+    bool field = {|FixAllInDocument:EXPRESSION|};
+    bool Method() => EXPRESSION;
+    bool Prop1 => EXPRESSION;
+    bool Prop2 { get } = EXPRESSION;
+    void If() { if (EXPRESSION) ; }
+    void Argument1() => Test(EXPRESSION);
+    void Argument2() => Test(() => EXPRESSION);
+    void Argument3() => Test(_ => EXPRESSION);
+    void For() { for (; EXPRESSION; ); }
+    void Local() { var local = EXPRESSION; }
+    void Conditional() { _ = EXPRESSION ? true : false; }
+    void Assignment() { _ = EXPRESSION; }
+    void Do() { do ; while (EXPRESSION); }
+    void While() { while (EXPRESSION) ; }
+    bool When() => o switch { _ when EXPRESSION => EXPRESSION };
+    bool Return() { return EXPRESSION; }
+    IEnumerable<bool> YieldReturn() { yield return EXPRESSION; }
+    Func<object, bool> SimpleLambda() => o => EXPRESSION;
+    Func<bool> ParenthesizedLambda() => () => EXPRESSION;
+    int i;
+    object o;
+}
+";
+
+        [InlineData("i == 0")]
+        [InlineData("i > 0")]
+        [InlineData("o is C")]
+        [InlineData("o is C c")]
+        [InlineData("o != null")]
+        [InlineData("!(o is C c)")]
+        [InlineData("!(o is null)")]
```

Comment:
I excluded this one because currently we don't offer "partial" fixes. e.g. `if (xyz && !(x is null))` remains intact. So I though it's better to support it as another analyzer dedicated to null checks.